src/Tizen.Network.Bluetooth/Tizen.Network.Bluetooth/BluetoothHidDevice.cs

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2018 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Threading.Tasks;
+using System.ComponentModel;
+
+namespace Tizen.Network.Bluetooth
+{
+    /// <summary>
+    /// A class which is used to provide the HID Device role.
+    /// </summary>
+    /// <remarks>
+    /// In HID profile, there are two roles Host and Device.
+    /// The Host(BluetoothHid) is a device that uses or requests the service of a HID.
+    /// The Device(BluetoothHidDevice) is a device that provides the service of human data input/output to/from the host.
+    /// </remarks>
+    /// <since_tizen> 6 </since_tizen>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class BluetoothHidDevice : BluetoothProfile
+    {
+        private TaskCompletionSource<BluetoothError> _taskForConnection;
+        private TaskCompletionSource<BluetoothError> _taskForDisconnection;
+
+        internal BluetoothHidDevice()
+        {
+            BluetoothHidDeviceImpl.Instance.ConnectionStateChanged += (s, e) =>
+            {
+                if (e.Address == RemoteAddress)
+                {
+                    if (_taskForConnection != null && !_taskForConnection.Task.IsCompleted)
+                    {
+                        _taskForConnection.SetResult((BluetoothError)e.Result);
```

Comment:
How about change return type of ConnectAsync to Task<bool>? because error should notified using by exception.
``` c#
 if (e.Result == BluetoothError.None || e.Result == BluetoothError.AlreadyDone) {
     _taskForConnection.SetResult(true);
 } else {
     _taskForConnection.SetException(BluetoothErrorFactory.CreateException(e.Result)); // you need to create a new factory method
 }
```

Reply:
Done. Thank you.